ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+Onboarding.swift

```diff
@@ -95,36 +88,99 @@ extension BrowserViewController {
+          }
+        }
+      )
+    } else {
+      if Preferences.FocusOnboarding.urlBarIndicatorShowBeShown.value {
+        Preferences.FocusOnboarding.urlBarIndicatorShowBeShown.reset()
+
+        controller = FocusNTPOnboardingViewController().then {
+          $0.setText(
+            title: Strings.FocusOnboarding.urlBarIndicatorTitle,
+            details: Strings.FocusOnboarding.urlBarIndicatorDescription
+          )
+        }
 
-        Preferences.FullScreenCallout.omniboxCalloutCompleted.value = true
-        Preferences.AppState.isOnboardingActive.value = false
+        presentNTPURLBarPopover(
+          controller: controller,
+          onDismiss: { [weak self] in
+            guard let self = self else { return }
+            self.triggerPromotedInAppPurchase(savedPayment: self.iapObserver.savedPayment)
+          },
+          onClickURLBar: { [weak self] in
+            guard let self = self else { return }
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+              self.topToolbar.tabLocationViewDidTapLocation(self.topToolbar.locationView)
+            }
+          }
+        )
+      }
+    }
 
-        self.triggerPromotedInAppPurchase(savedPayment: self.iapObserver.savedPayment)
-      },
-      didClickBorderedArea: { [weak self] in
-        guard let self = self else { return }
+    func presentNTPURLBarPopover(
+      controller: UIViewController & PopoverContentComponent,
+      onDismiss: @escaping () -> Void,
+      onClickURLBar: @escaping () -> Void
+    ) {
+      presentPopoverContent(
+        using: controller,
+        with: frame,
+        cornerRadius: topToolbar.locationContainer.layer.cornerRadius,
+        didDismiss: {
+          Preferences.FullScreenCallout.omniboxCalloutCompleted.value = true
+          Preferences.AppState.shouldDeferPromotedPurchase.value = false
+
+          onDismiss()
+        },
+        didClickBorderedArea: {
+          Preferences.FullScreenCallout.omniboxCalloutCompleted.value = true
+          Preferences.AppState.shouldDeferPromotedPurchase.value = false
+
+          onClickURLBar()
+        }
+      )
+    }
+  }
 
-        Preferences.FullScreenCallout.omniboxCalloutCompleted.value = true
-        Preferences.AppState.isOnboardingActive.value = false
+  private func addNTPTutorialPage() {
+    // The new onboarding will be only JP Region and this is part of old onboarding
+    guard Locale.current.regionCode != "JP" else {
+      return
+    }
 
-        DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
-          self.topToolbar.tabLocationViewDidTapLocation(self.topToolbar.locationView)
-        }
-      }
-    )
+    // NTP Education screen should load after onboarding is finished and user is on locale JP
+    let (educationPermitted, url) = (Locale.current.regionCode == "JP", URL.brave.ntpTutorialPage)
+
+    if educationPermitted {
+      tabManager.addTab(
+        PrivilegedRequest(url: url) as URLRequest,
```

Comment:
What request is this loading? If it's not potentially an internal page or session restore, then it shouldn't be privileged.

Reply:
Yeah I didnt notice, this is a part of the code that moved, it is loading a Brave page explaining for Japan users what is NTP. Request fixed at https://github.com/brave/brave-core/pull/22800/commits/49e062b086e60f8faf9ab30529d76ba879888445